package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +134,164 @@ def __init__(self, filename, **kwargs):
+            raise TypeError("No `atoms` attribute in object of type {}, "
+                            "please use a valid AtomGroup or Universe".format(
+                                type(obj))) from None
+
+        try:
+            elements = ag.elements
+        except NoDataError:
+            raise AttributeError(
+                "The `elements` attribute is required for the RDKitConverter "
+                "but is not present in this AtomGroup. Please refer to the "
+                "documentation to guess elements from other attributes or "
+                "type `help(mda.topology.guessers)`") from None
+
+        # attributes accepted in PDBResidueInfo object
+        pdb_attrs = {}
+        for attr in RDATTRIBUTES.keys():
+            if hasattr(ag, attr):
+                pdb_attrs[attr] = getattr(ag, attr)
+        # others
+        other_attrs = {}
+        for attr in ["bfactors", "charges", "segids", "types"]:
+            if hasattr(ag, attr):
+                other_attrs[attr] = getattr(ag, attr)
+
+        mol = Chem.RWMol()
+        atom_mapper = {}
+
+        for i, (atom, element) in enumerate(zip(ag, elements)):
+            # create atom
+            rdatom = Chem.Atom(element.capitalize())
+            # disable adding H to the molecule
+            rdatom.SetNoImplicit(True)
+            # add PDB-like properties
+            mi = Chem.AtomPDBResidueInfo()
+            for attr, values in pdb_attrs.items():
+                _add_mda_attr_to_rdkit(attr, values[i], mi)
+            rdatom.SetMonomerInfo(mi)
+            # other properties
+            for attr in other_attrs.keys():
+                value = other_attrs[attr][i]
+                attr = attr[:-1] # plural to singular
+                if isinstance(value, np.float):
+                    rdatom.SetDoubleProp("_MDAnalysis_%s" % attr, float(value))
+                elif isinstance(value, np.int):
+                    rdatom.SetIntProp("_MDAnalysis_%s" % attr, int(value))
+                else:
+                    rdatom.SetProp("_MDAnalysis_%s" % attr, value)
+            rdatom.SetIntProp("_MDAnalysis_index", int(atom.ix))
+            # add atom
+            index = mol.AddAtom(rdatom)
+            # map index in universe to index in mol
+            atom_mapper[atom.ix] = index
+
+        try:
+            bonds = ag.bonds
+            if (len(bonds) == 0) and (ag.n_atoms > 1):
+                # force guessing bonds
+                raise NoDataError
+        except NoDataError:
+            warnings.warn(
+                "No `bonds` attribute in this AtomGroup. Guessing bonds based"
+                "on atoms coordinates")
+            ag.guess_bonds()
+            bonds = ag.bonds
+
+        for bond in bonds:
+            try:
+                bond_indices = [atom_mapper[i] for i in bond.indices]
+            except KeyError:
+                # one of the atoms of the bond is not part of the atomgroup
+                continue
```

Comment:
Not sure if this is an issue or not: 
The KeyError is raised when one of the atom in bond.indices wasn't seen when converting MDA atoms to RDKit, and happens when selecting a fragment of a molecule (`u.select_atoms("resid 2").convert_to("RDKIT")` for example). 
Which means that this selection (resid2) will contain bonds between atoms of resid2, but also the bond between resid1 and the Nterm of resid2, and the bond between the Cterm of resid2 and resid3, although resid1 and resid3 are not part of the selection. 
Is this intended ? Or should the selection only return bonds where both atoms of the bond belong to the selection ?

Reply:
moved this to #2821